src/OmniSharp.Abstractions/Services/IProjectSystem.cs

```diff
@@ -29,5 +29,11 @@ public interface IProjectSystem
         /// <param name="filePath">The file path to the project to retrieve. Alternatively,
         /// a file path to a document within a proejct may be specified.</param>
         Task<object> GetProjectModelAsync(string filePath);
+
+        /// <summary>
+        /// This project system will try to locate projects that reference given file and will wait until the projects and their references are fully loaded.
+        /// </summary>
+        /// <param name="filePath">File for which to locate and load projects.</param>
+        Task WaitForProjectsToLoadForFileAsync(string filePath);
```

Comment:
@rchande @akshita31 @DustinCampbell : myself and couple of folks were dogfooding the changes and so far everything works as expected (within the scenarios described in https://github.com/OmniSharp/omnisharp-roslyn/issues/1316). I'm moving towards beefing up UTs and appreciate if you could share any additional concerns you might have about the approach. Thanks!